fem/fe_coll.hpp

```diff
@@ -1305,6 +1305,60 @@ class Local_FECollection : public FiniteElementCollection
    virtual int GetContType() const { return DISCONTINUOUS; }
 };
 
-}
+/** @brief Radial basis function collection.
+    Instead of using polynomials as a basis, these methods use
+    Gaussian-like functions centered at chosen points. Reproducing
+    kernels augment the radial basis functions so that the resultant
+    basis can exactly represent functions of a chosen polynomial degree. */
+class KernelFECollection : public FiniteElementCollection
+{
+private:
+   int maxDim;
+   char d_name[32];
+   ScalarFiniteElement *Tr_Elements[Geometry::NumGeom];
+   ScalarFiniteElement *L2_Elements[Geometry::NumGeom];
+   int *SegDofOrd[2]; // for rotating segment dofs in 1D
+   int *OtherDofOrd;
+   bool ValidGeomType(Geometry::Type GeomType) const;
+
+public:
+   /** @brief Create a radial basis function or reproducing kernel finite
+       element collection.
+       @param D           Reference space dimension
+       @param numPointsD  Number of points across the element in each dimension
+       @param rbfType     Type of radial basis function, from RBFType
+       @param order       Order of polynomial correction, >= 0 for RK or -1 for RBF
+       @param h           Shape parameter, approximately equal to the number of points in the support radius in one dimension, should generally be > order
+       @param distNorm    Norm used for distance, usually 2 = Euclidean distance
+       @param intOrder    Number of integration points per RBF point in each dimension
+       @param faceFactor  1.0 = points end on face, 0.0 = points end at dx/2 from face
```

Comment:
This is confusing, please explain it properly.
Are the points always equidistant?
If there is one point in total, which face does it end on?
Value of `0.0` seems more logical to be on the face.
Do you really need such a messy procedure to generate equidistant points?

Reply:
The points are always equidistant. If faceFactor is set to 0.0, then the points are also equidistant with those in surrounding elements in reference space. If it is set to 1.0, then there are points on the faces but the points remain equidistant. 

The RBFFiniteElement enforces at least two points in each dimension to keep some level of accuracy, so the issue with one point doesn't show up.

I expect that most of the time, users will either be interested in points on the face or dx/2 from the boundary. I left this open to increase the choices, but if we wanted to simplify things, I could make it a bool. Any thoughts on how I could make it clearer or on a better name for this parameter?

Reply:
Yes I think an `enum` with 2 options will be simpler. We usually say `ClosedUniform` and `OpenUniform` spacing, see for example `class Quadrature1D`.